src/Driver/ChromeDriver/VersionResolver.php

```diff
@@ -74,7 +77,32 @@ public function latest(): Version
 
     public function supports(Browser $browser): bool
     {
-        return $browser->name()->equals(BrowserName::GOOGLE_CHROME()) || $browser->name()->equals(BrowserName::CHROMIUM());
+        $browserName = $browser->name();
+
+        return $browserName->equals(BrowserName::GOOGLE_CHROME()) || $browserName->equals(BrowserName::CHROMIUM());
+    }
+
+    /**
+     * @throws RedirectionExceptionInterface
+     * @throws ClientExceptionInterface
+     * @throws TransportExceptionInterface
+     * @throws ServerExceptionInterface
+     */
+    private function getVersionString(Browser $browser): string
+    {
+        if ($browser->version()->major() < self::MAJOR_VERSION_ENDPOINT_BREAKPOINT) {
+            return $this->httpClient->request('GET', $this->getBrowserVersionEndpoint($browser))->getContent();
+        }
+
+        $response = $this->httpClient->request('GET', self::VERSION_ENDPOINT_JSON);
+        $versions = $response->toArray();
+        if (! array_key_exists($browser->version()->toString(), $versions['builds'])) {
+            throw new Unsupported(
+                sprintf('There is no build for version : %s', $browser->version()->toString())
+            );
```

Comment:
**Suggestion:** use UnexpectedValueException instead of Unsupported.
The Unsupported is intended for browser-specific support.

Reply:
done